core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -1065,22 +1058,32 @@ impl RecoverableVertexStore for RocksDBStore {
 }
 
 fn encode_to_rocksdb_bytes(partition_key: &DbPartitionKey, sort_key: &DbSortKey) -> Vec<u8> {
-    let mut buffer = Vec::with_capacity(1 + partition_key.0.len() + sort_key.0.len());
+    let mut buffer = Vec::with_capacity(2 + partition_key.node_key.len() + sort_key.0.len());
```

Comment:
```suggestion
    let mut buffer = Vec::with_capacity(1 + partition_key.node_key.len() + 1 + sort_key.0.len());
```

(or some `size_of()`, or some well-named constants - and the same numbers can be used inside `decode_from_rocksdb_bytes()`)

But this is nitpicking only for my own readability; can be done in some next PR.